apps/server/lib/lexical/server/window.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Lexical.Server.Window do
+  alias Lexical.Protocol.Notifications.LogMessage
+  alias Lexical.Protocol.Notifications.ShowMessage
+  alias Lexical.Server.Transport
+
+  @type level :: :error | :warning | :info | :log
+
+  @spec log(level, String.t()) :: String.t()
+  def log(level, message) when level in [:error, :warning, :info, :log] do
+    log_message = apply(LogMessage, level, [message])
+    Transport.write(log_message)
+    message
```

Comment:
We probably should return the result of Transoort.write, or just ok for both log and show. I think returning the message made more sense when it could be anything rather than just a string